operator/cilium_node.go

```diff
@@ -370,6 +370,21 @@ func runCNPNodeStatusGC(name string, clusterwide bool, nodeStore cache.Store) {
 					}
 
 					for _, cnp := range cnpItemsList {
+						// If CNP status updates have been disabled, we clean all the
+						// possible updates written when the option was still enabled.
+						// This is done to avoid the stale updates to accumulate and thus
+						// hinder scalability.
+						if option.Config.DisableCNPStatusUpdates {
+							wg.Add(1)
+							cnpCpy := cnp.DeepCopy()
```

Comment:
Because the `cnpCpy` is never modified inside the `clearStatusNodesInCnp` and because `cnp` is not being retrieved from the k8s-client caches, we don't need to perform a `DeepCopy()` here.